game-backend/src/game-engine/state-selectors/index.ts

```diff
@@ -0,0 +1,4 @@
+import selectFoods from './food-selector';
+import selectUserUnit from './userUnit-selector';
+
+export { selectFoods, selectUserUnit };
```

Comment:
Just as a small hint:
We already have `helpers/gameObjects.ts` which contains very similar logic. Would it make sense to combine these helpers?

Reply:
Yes it would. To accomplish this, should I move the folder state-selectors to helpers or integrate the files directly?

Reply:
Both solutions would be feasible. I prefer to keep the `state-selectors` and use the generic Object Helpers inside of them. (See latest commit)
There is only one open point: Should the unit test be refactored to test `helper/gameObject.ts` instead of the food selector? (I'd advocate to refactor it to `gameObject.test.ts`)

Reply:
What if we leave the state-selector tests as is and add some tests to gameObject.ts. With this approach we could ensure that if someone changes gameObjects.ts we can be sure that the state-selectors still work

Reply:
Then we test the same functionality twice.
Imo the `state-selector.test` must only test what he's doing (arguably nothing at all).

Or what test cases would you put in there?

Reply:
I see your point. So I move and adjust the tests in `food-selector.test.ts` to test `getGameObjectsPerType()` and later if the food-selector maybe gets more advanced we write some tests for it.